src/protagonist/API/Features/NamedQueries/NamedQueriesController.cs

```diff
@@ -0,0 +1,166 @@
+﻿using API.Features.NamedQueries.Converters;
+using API.Features.NamedQueries.Requests;
+using API.Infrastructure;
+using API.Settings;
+using DLCS.Core.Collections;
+using DLCS.HydraModel;
+using DLCS.Web.Auth;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace API.Features.NamedQueries;
+
+/// <summary>
+/// DLCS REST API Operations for Named Queries
+/// </summary>
+[Route("/customers/{customerId}/namedQueries")]
+[ApiController]
+public class NamedQueriesController : HydraController
+{
+    public NamedQueriesController(
+        IMediator mediator,
+        IOptions<ApiSettings> options) : base(options.Value, mediator)
+    {
+    }
+    
+    /// <summary>
+    /// Get a list of all available Named Queries, either global or owned by the user
+    /// </summary>
+    /// <returns>HydraCollection of NamedQuery</returns>
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetNamedQueries(
+        [FromRoute] int customerId,
+        CancellationToken cancellationToken)
+    {
+        var namedQueries = new GetAllNamedQueries(customerId);
+        
+        return await HandleListFetch<DLCS.Model.Assets.NamedQueries.NamedQuery, GetAllNamedQueries, NamedQuery>(
+            namedQueries,
+            nq => nq.ToHydra(GetUrlRoots().BaseUrl),
```

Comment:
Do the `id` values for global named queries have the calling customers id in them? If so there may be a bug similar to https://github.com/dlcs/protagonist/issues/525

Reply:
Fetching global NQ's not addresses in the acceptance criteria for ticket - created #566 to cover this.